OneSignalSDK/onesignal/src/main/java/com/onesignal/OSTaskController.java

```diff
@@ -0,0 +1,169 @@
+package com.onesignal;
+
+import androidx.annotation.NonNull;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicLong;
+
+class OSTaskController {
+
+    static final String OS_PENDING_EXECUTOR = "OS_PENDING_EXECUTOR_";
```

Comment:
Just curious why we have an `_` at the end of the `OS_PENDING_EXECUTOR_`

Reply:
Oh that's because 

`newThread.setName(OS_PENDING_EXECUTOR + newThread.getId());`